Yesod/Auth/OAuth2/Nylas.hs

```diff
@@ -0,0 +1,93 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Yesod.Auth.OAuth2.Nylas
+    ( oauth2Nylas
+    , module Yesod.Auth.OAuth2
+    ) where
+
+#if __GLASGOW_HASKELL__ < 710
+import Control.Applicative ((<$>), (<*>))
+#endif
+
+import Control.Monad (mzero)
+import Control.Exception.Lifted (throwIO)
+import Data.Aeson (FromJSON, Value(..), parseJSON, decode, (.:))
+import Data.Monoid ((<>))
+import Data.Text (Text)
+import Data.Text.Encoding (encodeUtf8, decodeUtf8)
+import Data.Vector ((!?))
+import Network.HTTP.Client (applyBasicAuth, parseUrl, httpLbs, responseStatus, responseBody)
+import Network.HTTP.Conduit (Manager)
+
+import Yesod.Auth
+import Yesod.Auth.OAuth2
+
+import qualified Data.Text as T
+import qualified Network.HTTP.Types as HT
+
+data NylasNamespace = NylasNamespace
+    { nylasNamespaceId :: Text
+    , nylasNamespaceAccountId :: Text
+    , nylasNamespaceEmailAddress :: Text
+    , nylasNamespaceName :: Text
+    , nylasNamespaceProvider :: Text
+    , nylasNamespaceOrganizationUnit :: Text
+    }
+
+instance FromJSON NylasNamespace where
+    parseJSON (Array singleton) = case singleton !? 0 of
+        Just (Object o) -> NylasNamespace
+            <$> o .: "id"
+            <*> o .: "account_id"
+            <*> o .: "email_address"
+            <*> o .: "name"
+            <*> o .: "provider"
+            <*> o .: "organization_unit"
+        _ -> mzero
+    parseJSON _ = mzero
+
+oauth2Nylas :: YesodAuth m
+            => Text -- ^ Client ID
```

Comment:
Thoughts on making a `type` alias for `ClientID`/`ClientSecret` to replace these comments?